### ISSUE TITLE
Display of PNOR FW version from Boston nodes

### ISSUE THREAD

**Issue by gurevichmark** [xcat2/xcat-core#6264]
The display of PNOR FW version from Boston nodes appears to be incorrect:
```
[root@stratton01 xCAT_plugin]# rflash boston32,f5u25 -c
 boston32: Node firmware version for SMC BMC FW   component:   2.04.0
 boston32: Node firmware version for PNOR FW      component:   0.00.17307680
 boston32: Node firmware version for  component:   0.00.0
 f5u25: Node firmware version for BOOT component:   2.13.104536
 f5u25: Node firmware version for APP component:   2.13.104536
 f5u25: Node firmware version for BIOS component:   1.12.85
```

Related issue #4615

**Comment by gurevichmark** [comment]
Tried various boston nodes that we have to see the difference between what `rflash` and `rinv` returns.
It seems `BMC FW` versions returned by  `rflash` and `rinv` match, but `PNOR FW` and `System Firmware Product Version` do not. 
Are `PNOR FW` and `System Firmware Product Version` supposed to be the same ?
How can we verify what `PNOR FW` version supposed to be ?

```
[root@stratton01 ~]# rflash boston02 -c; echo " ";rinv boston02 firm | egrep "BMC|Version"
boston02: Node firmware version for SMC BMC FW   component:   0.36.0
boston02: Node firmware version for PNOR FW      component:  10.30.5920
boston02: Node firmware version for  component:   0.00.0

boston02: BMC Firmware: 0.36
boston02: System Firmware Product Version: open-power-SUPERMICRO-P9DSU-0.19-20171030C
```
```
[root@stratton01 ~]# rflash boston30 -c; echo " ";rinv boston30 firm | egrep "BMC|Version"
boston30: Node firmware version for SMC BMC FW   component:   1.13.0
boston30: Node firmware version for PNOR FW      component:   1.30.6176
boston30: Node firmware version for  component:   0.00.0

boston30: BMC Firmware: 1.13
boston30: System Firmware Product Version: open-power-SUPERMICRO-P9DSU-0.24-20180130
```
```
[root@stratton01 ~]# rflash boston32 -c; echo " ";rinv boston32 firm | egrep "BMC|Version"
boston32: Node firmware version for SMC BMC FW   component:   2.04.0
boston32: Node firmware version for PNOR FW      component:   0.00.17307680
boston32: Node firmware version for  component:   0.00.0

boston32: BMC Firmware: 2.04
boston32: System Firmware Product Version: open-power-SUPERMICRO-P9DSU-0.24.2-20180801
```
```
[root@stratton01 ~]# rflash boston36 -c; echo " ";rinv boston36 firm | egrep "BMC|Version"
boston36: Node firmware version for SMC BMC FW   component:   1.23.0
boston36: Node firmware version for PNOR FW      component:   8.01.6176
boston36: Node firmware version for  component:   0.00.0

boston36: BMC Firmware: 1.23
boston36: System Firmware Product Version: open-power-SUPERMICRO-P9DSU-0.24.2-20180801
```
```
[root@stratton01 ~]# rflash boston38 -c; echo " ";rinv boston38 firm | egrep "BMC|Version"
boston38: Node firmware version for SMC BMC FW   component:   1.23.0
boston38: Node firmware version for PNOR FW      component:   8.01.6176
boston38: Node firmware version for  component:   0.00.0

boston38: BMC Firmware: 1.23
boston38: System Firmware Product Version: open-power-SUPERMICRO-P9DSU-0.24.2-20180801
[root@stratton01 ~]#
```

**Comment by gurevichmark** [comment]
@zet809 Do you know why to get PNOR FW info we use `raw` commands like:
```
boston02: [stratton01]: [ipmi_debug] rflash: check(-c -V), raw_cmd: netfn(0x2c=>), cmd(0x2f=>), data=[0, 1, 1]
boston02: [stratton01]: [ipmi_debug] rflash: check(-c -V), raw_cmd: netfn(0x2c=>), cmd(0x2f=>), data=[0, 1, 2]
boston02: [stratton01]: [ipmi_debug] rflash: check(-c -V), raw_cmd: netfn(0x2c=>), cmd(0x2f=>), data=[0, 2, 1]
boston02: [stratton01]: [ipmi_debug] rflash: check(-c -V), raw_cmd: netfn(0x2c=>), cmd(0x2f=>), data=[0, 2, 2]
```
Instead of just `ipmitool-xcat -H boston02-bmc -U ADMIN -P ADMIN fru print 47` 
which would return ` open-power-SUPERMICRO-P9DSU-0.19-20171030C`

The data returned by `raw 0x2c` commands seems to be different from what `fru 47` returns.

Compare `boston32` and `boston36` output above. 
Both have the same `System Firmware Product Version: open-power-SUPERMICRO-P9DSU-0.24.2-20180801` returned by `rinv` but different 
`Node firmware version for PNOR FW      component:   0.00.17307680` and 
`Node firmware version for PNOR FW      component:   8.01.6176` returned by `rflash`

**Comment by gurevichmark** [comment]
Doing some more testing of our machines.

One Firestone machines seem to agree on FW version:
```
[root@stratton01 ~]# rflash c910f3u27-fs3 -c; echo " ";rinv c910f3u27-fs3 firm | egrep "BMC|Version"
c910f3u27-fs3: Node firmware version for BOOT component:   2.13.104530
c910f3u27-fs3: Node firmware version for APP component:   2.13.104530
c910f3u27-fs3: Node firmware version for BIOS component:   1.12.70

c910f3u27-fs3: BMC Firmware: 2.13
c910f3u27-fs3: System Firmware Product Version: IBM-firestone-ibm-OP8_v1.12_2.70
[root@stratton01 ~]#
```
But another Firestone does not:
```
[root@stratton01 ~]# rflash c910f3u29-fs1 -c; echo " ";rinv c910f3u29-fs1 firm | egrep "BMC|Version"
c910f3u29-fs1: Node firmware version for BOOT component:   2.00.79228
c910f3u29-fs1: Node firmware version for APP component:   2.08.83922
c910f3u29-fs1: Node firmware version for BIOS component:   0.00.0

c910f3u29-fs1: BMC Firmware: 2.08
c910f3u29-fs1: System Firmware Product Version: IBM-firestone-ibm-OP8_PFD_v1.4_0.9
[root@stratton01 ~]#
```
Garrison does not match:
```
[root@stratton01 ~]# rflash f5u25 -c; echo " ";rinv f5u25 firm | egrep "BMC|Version"
f5u25: Node firmware version for BOOT component:   2.13.104536
f5u25: Node firmware version for APP component:   2.13.104536
f5u25: Node firmware version for BIOS component:   1.12.85

f5u25: BMC Firmware: 2.13
f5u25: System Firmware Product Version: IBM-garrison-ibm-OP8_v1.11_2.19
[root@stratton01 ~]# 
```

Another mismatch:
```
[root@stratton01 ~]# rflash briggs01 -c; echo " ";rinv briggs01 firm | egrep "BMC|Version"
briggs01: Node firmware version for SMC BMC FW   component:   1.27.0
briggs01: Node firmware version for PNOR FW      component:   1.05.6176
briggs01: Node firmware version for  component:   0.00.0

briggs01: BMC Firmware: 1.27
briggs01: System Firmware Product Version: open-power-IBM-P8DTU-V2.00.GA2.SP1-20180105-prod
[root@stratton01 ~]#
```

**Comment by zet809** [comment]
hi, @gurevichmark , for `rflash -c` I'd like to keep it consistent with output of `ipmitool hpm check`
For example:
```
[root@stratton01 ~]# ipmitool-xcat -I lanplus -H boston32-bmc -U ADMIN -P ADMIN hpm check

PICMG HPM.1 Upgrade Agent 1.0.9:

-------Target Information-------
Device Id          : 0x20
Device Revision    : 0x1
Product Id         : 0x0985
Manufacturer Id    : 0x2a7c (Supermicro)


--------------------------------------------------------------------------
|ID  | Name        |                     Versions                        |
|    |             |     Active      |     Backup      |      Deferred   |
--------------------------------------------------------------------------
|*  0|SMC BMC FW   |   2.04 00000000 | ---.-- -------- | ---.-- -------- |
|*  1|PNOR FW      |   0.00 20180801 | ---.-- -------- | ---.-- -------- |
--------------------------------------------------------------------------
(*) Component requires Payload Cold Reset


[root@stratton01 ~]# ipmitool-xcat -I lanplus -H boston32-bmc -U ADMIN -P ADMIN fru print 47
 Product Name          : OpenPOWER Firmware
 Product Version       : open-power-SUPERMICRO-P9DSU-0.24.2-20180801
 Product Extra         : 	op-build-8c932cd
 Product Extra         : 	skiboot-v5.9.8-p20a103c
 Product Extra         : 	hostboot-89d9dff
 Product Extra         : 	linux-4.13.16-openpower1-p4183454
 Product Extra         : 	petitboot-v1.6.3-p2610636
 Product Extra         : 	machine-xml-bbf0d7a
 Product Extra         : 	occ-84f3564
 Product Extra         : 	sbe-99e2fe2
```

**Comment by gurevichmark** [comment]
@zet809 Following the recent email discussion about the "Boston PNOR FW version", it sounds like we should pull the PNOR FW version from what `fru 47` returns and not `hpm check` for `rflash -c`.

It seems pretty clear for Boston machines, but should we do it also for Firestone and Garrison ?

**Comment by zet809** [comment]
Even through we don't modify the `rflash -c` have same output with `rinv firm`, we still need to update it since it is different with `ipmitool hpm check`, @gurevichmark 
```
[root@stratton01 ~]# ipmitool-xcat -I lanplus -H boston32-bmc -U ADMIN -P ADMIN hpm check

PICMG HPM.1 Upgrade Agent 1.0.9:

-------Target Information-------
Device Id          : 0x20
Device Revision    : 0x1
Product Id         : 0x0985
Manufacturer Id    : 0x2a7c (Supermicro)


--------------------------------------------------------------------------
|ID  | Name        |                     Versions                        |
|    |             |     Active      |     Backup      |      Deferred   |
--------------------------------------------------------------------------
|*  0|SMC BMC FW   |   2.04 00000000 | ---.-- -------- | ---.-- -------- |
|*  1|PNOR FW      |   0.00 20180801 | ---.-- -------- | ---.-- -------- |
--------------------------------------------------------------------------
(*) Component requires Payload Cold Reset

[root@stratton01 ~]# rflash boston32 -c
boston32: Node firmware version for SMC BMC FW   component:   2.04.0
boston32: Node firmware version for PNOR FW      component:   0.00.17307680. ==> shall be 0.00.20180801
boston32: Node firmware version for  component:   0.00.0
```

**Comment by gurevichmark** [comment]
PR #6277 fixes this issue - Undo the changes made PR #4625. At that time it was believed that output of `rflash -c` should match contents of `/proc/ractrends/Helper/FwInfo` file. But now we are not sure of contents of that file are correct, and the output should match `ipmitool hpm check`.

**Comment by zet809** [comment]
Thx, @gurevichmark , I have merge the PR #6277 , so close this issue.
